File: src/main/java/model/entities/Alunos.java
package model.entities;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.Objects;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "alunos")
public class Alunos implements Serializable {
	private static final long serialVersionUID = 1L;
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	private String nome;
	private String cpf;
	private String celular;
	private LocalDate aniversario;
	
	public Alunos() {}
	public Alunos(Integer id, String nome, String cpf, String celular, LocalDate aniversario) {
		this.id = id;
		this.nome = nome;
		this.cpf = cpf;
		this.celular = celular;
		this.aniversario = aniversario;
	}
	
	public Integer getId() {
		return id;
	}
	public void setId(Integer id) {
		this.id = id;
	}
	
	public String getNome() {
		return nome;
	}
	public void setNome(String nome) {
		this.nome = nome;
	}
	
	public String getCpf() {
		return cpf;
	}
	public void setCpf(String cpf) {
		this.cpf = cpf;
	}
	
	public String getCelular() {
		return celular;
	}
	public void setCelular(String celular) {
		this.celular = celular;
	}
	
	public LocalDate getAniversario() {
		return aniversario;
	}
	public void setAniversario(LocalDate aniversario) {
		this.aniversario = aniversario;
	}
	
	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Alunos other = (Alunos) obj;
		return Objects.equals(id, other.id);
	}
	
	@Override
	public String toString() {
		return "Informações do aluno \n" + "id: " + id + "\n" + "Nome: " + nome + "\n" + "cpf: " + cpf + "\n"
				+ "Celular: " + celular + "\n" + "Data de nascimento: " + aniversario;
	}
}

File: src/models/entities/Alunos.js
export default class Alunos {
    #id;
    #nome;
    #cpf;
    #celular;
    #aniversario;

    constructor(id, nome, cpf, celular, aniversario) {
        this.#id = id;
        this.nome = nome;
        this.cpf = cpf;
        this.celular = celular;
        this.aniversario = aniversario;
    }

    get id() {
        return this.#id;
    }

    get nome() {
        return this.#nome;
    }
    set nome(valor) {
        this.#nome = valor;
    }

    get cpf() {
        return this.#cpf;
    }
    set cpf(valor) {
        this.#cpf = valor;
    }

    get celular() {
        return this.#celular;
    }
    set celular(valor) {
        this.#celular = valor;
    }

    get aniversario() {
        return this.#aniversario;
    }
    set aniversario(valor) {
        this.#aniversario = valor;
    }
    
    toJSON() {
    return {
        id: this.#id,
        nome: this.#nome,
        cpf: this.#cpf,
        celular: this.#celular,
        aniversario: this.#aniversario
    };
}
}